src/EFCore.PG/Utilities/Check.cs

```diff
@@ -115,6 +115,19 @@ public static string NullButNotEmpty([CanBeNull] string value, [InvokerParameter
             return value;
         }
 
+        public static IReadOnlyCollection<T> NullButNotEmpty<T>([CanBeNull] IReadOnlyCollection<T> value, [InvokerParameterName] [NotNull] string parameterName)
```

Comment:
nit: Here we can restrict the value type to `IReadOnlyList<T>` as it's done below. Otherwise, we should review other guards, but in other PR.

Reply:
`IReadOnlyCollection` is more general than `IReadOnlyCollection`. We only need the `Count` property, not random acces, why restrict it more?

Reply:
Absolutely agree with you, but other checks use lists. Personally I think that we should review our code and change it. So you could leave this as is (:

Reply:
Just to say that Check.cs is a file copied as-is from EF Core, and it's probably best to not touch this kind of thing - it makes it a bit more difficult to track upstream changes (e.g. diffs).